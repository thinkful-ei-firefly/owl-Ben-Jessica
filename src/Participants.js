import React from 'react';
import Person from './Person';

function Participants(props) {
  const list = props.participants.map(person => {
    return (
      <Person
        key={person.id}
        avatar={person.avatar}
        name={person.name}
        inSession={person.inSession}
        onStage={person.onStage}
      />
    );
  });
  return <div className="participants">{list}</div>;
}

export default Participants;
