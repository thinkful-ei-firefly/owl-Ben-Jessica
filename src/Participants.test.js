import React from 'react';
import ReactDOM from 'react-dom';
import store from './store'
import renderer from 'react-test-renderer';

import Participants from './Participants';

describe('<Stage />', () => {
    const part = store.participants;
    
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<Participants participants={part} />, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        // Render the component, as JSON
        const tree = renderer.create(<Participants participants={part} />).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});