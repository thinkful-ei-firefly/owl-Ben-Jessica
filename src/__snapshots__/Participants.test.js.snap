// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage /> renders this UI as expected 1`] = `
<div
  className="participants"
>
  <div>
    <img
      alt="icon"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <span
      className="personName"
    >
      Koren Templeton
    </span>
    <span
      className="green"
    >
      ⬤
    </span>
    <span>
      on stage
    </span>
  </div>
  <div>
    <img
      alt="icon"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <span
      className="personName"
    >
      Caty Flucker
    </span>
    <span
      className="green"
    >
      ⬤
    </span>
    <span>
      in session
    </span>
  </div>
  <div>
    <img
      alt="icon"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <span
      className="personName"
    >
      Axe Kubicka
    </span>
    <span
      className="gray"
    >
      ⬤
    </span>
    <span>
      left session
    </span>
  </div>
  <div>
    <img
      alt="icon"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <span
      className="personName"
    >
      Frank Runciman
    </span>
    <span
      className="green"
    >
      ⬤
    </span>
    <span>
      in session
    </span>
  </div>
  <div>
    <img
      alt="icon"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <span
      className="personName"
    >
      Ashla Attwool
    </span>
    <span
      className="green"
    >
      ⬤
    </span>
    <span>
      on stage
    </span>
  </div>
</div>
`;
